src/strategies/subgraph-split-delegation/index.ts

```diff
@@ -0,0 +1,174 @@
+
+  const allAddresses = new Set(addresses);
+
+  members.forEach((member) => {
+    allAddresses.add(member.address);
+    member.delegators?.forEach((delegation) =>
+      delegation.delegator
+        ? allAddresses.add(delegation.delegator.address)
+        : null
+    );
+    member.delegatees?.forEach((delegation) =>
+      delegation.delegatee
+        ? allAddresses.add(delegation.delegatee.address)
+        : null
+    );
+  });
+
+  const scores: { [k: string]: unknown }[] = (
+    await getScoresDirect(
+      space,
+      options.strategies,
+      network,
+      provider,
+      [...allAddresses],
+      snapshot
+    )
+  ).filter((score) => Object.keys(score).length !== 0);
+
+  return Object.fromEntries(
+    addresses.map((address) => {
+      const member = members.find(
+        (member) => member.address.toLowerCase() === address.toLowerCase()
+      );
+      return [
+        getAddress(address),
+        member ? getVp(member, scores) : getAddressScore(scores, address)
+      ];
+    })
+  );
+}
+
+const getAddressScore = (
+  scores: { [k: string]: any }[],
+  address?: string
+): any => {
+  if (!address) return 0;
+  return scores.reduce((total, score) => total + (score[address] ?? 0), 0);
+};
+
+const getVp = (member: Member, scores: { [k: string]: any }[]): any => {
+  const addressScore = getAddressScore(scores, member.address);
+  const delegatedVp =
+    member.delegatees?.reduce((total, delegation) => {
+      const vp = addressScore;
+      return total + delegation.ratio * vp;
+    }, 0) ?? 0;
+  const receivedVp =
+    member.delegators?.reduce((total, delegation) => {
+      const vp = getAddressScore(scores, delegation.delegator?.address);
+      return total + delegation.ratio * vp;
+    }, 0) ?? 0;
+
+  return addressScore + receivedVp - delegatedVp;
+};
+
+async function subgraphRequestWithPagination(
+  subgraphURL,
+  addresses,
+  space,
+  block
+): Promise<Member[]> {
+  const chunkSize = 500;
+  const chunks: string[][] = [];
+  for (let i = 0; i < addresses.length; i += chunkSize) {
+    chunks.push(addresses.slice(i, i + chunkSize));
+  }
+
+  const results: Member[] = [];
+  for (const chunk of chunks) {
+    const params = {
+      members: {
+        __args: {
+          block: { number: block.number },
+          where: {
+            address_in: chunk.map((address) => address.toLowerCase())
+          }
+        },
+        id: true,
+        address: true,
+        delegators: {
+          __args: {
+            where: {
+              context: space,
+              expirationTimestamp_gte: block.timestamp
+            }
+          },
+          delegator: {
+            address: true
+          },
+          ratio: true
+        },
+        delegatees: {
+          __args: {
+            where: {
+              context: space,
+              expirationTimestamp_gte: block.timestamp
+            }
+          },
+          delegatee: {
+            address: true
```

Comment:
Not too sure, but i think if a address has more than 1000 delegations this may ignore them?

Reply:
If the chuck of addresses have more than 1000 delegators

Reply:
Yeah, I think you are right. I'm going to paginate the delegations too. Thanks!